sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala

```diff
@@ -847,6 +847,8 @@ private[hive] trait HiveInspectors {
 
         ObjectInspectorFactory.getStandardConstantMapObjectInspector(keyOI, valueOI, jmap)
       }
+    case Literal(_, dt: StructType) =>
+      toInspector(dt)
```

Comment:
The fix looks correct.

However, to use `toInspector` here is to invoke `toInspector(dataType: DataType)` for nested fields. For the fields, not constant object inspector is used anymore.

Can we follow array and map literals logics, to use constant object inspector for nested fields?

Reply:
Looking at the logics of array and map literals , there is no constant object inspector for nested fields.

It seems that hive udf only requires the object to be ```instanceof ConstantObjectInspector``` , no requirement for nested fields.

Hive implements ```ObjectInspectorFactory#getStandardConstantStructObjectInspector``` on [HIVE-11428](https://issues.apache.org/jira/browse/HIVE-11428), but there is no such method in hive-exec 1.2.1.spark2.

https://github.com/apache/hive/blob/d72f70ad959c27a713f53d5e8b9172d662a29d51/serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/ObjectInspectorUtils.java#L1422-L1459

Reply:
Hmm, I thought to fill constant object inspectors for struct fields into `ObjectInspectorFactory.getStandardStructObjectInspector`. But there is no `getStandardConstantStructObjectInspector` for null value case.

Ok. Then I think current fix is good. Thanks.